src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -659,24 +661,14 @@ private void BackgroundLogger()
                     {
                         Console.Error.WriteLine("ERROR! Out of memory exception while trying to log, will avoid logging for the next 5 seconds");
 
-                        var time = 5000;
-                        var current = Stopwatch.GetTimestamp();
+                        DisableLogsFor(threadStates, TimeSpan.FromSeconds(5));
+                    }
+                    catch (IOException ioe) when (IsOutOfDiskSpaceException(ioe))
```

Comment:
Why is this just for out of disk space?
Shouldn't we do that for pretty much everything? (No permission on drive, for example)

Reply:
done